pkgs/build-support/cc-wrapper/add-flags.sh

```diff
@@ -9,6 +9,7 @@ var_templates_list=(
     NIX+CFLAGS_LINK
     NIX+CXXSTDLIB_COMPILE
     NIX+CXXSTDLIB_LINK
+    NIX+GCC_ARCH_FLAGS
```

Comment:
I’d like to avoid adding more args here. Could we add it via /nix-support/cc-cflags?

Reply:
I do not know.
My only point was to land the flags to ```extraBefore```, so the "real" command line -march= could overwrite it (an example above: ```openblas``` builds CPU-optimized code for all CPUs include the higher models, so it need to invoke ```gcc -march=skylake-avx512``` to compile some files)

Reply:
I think it should be before. Also will avoid a mass rebuild I think.

Reply:
It doesn't appear so

https://github.com/NixOS/nixpkgs/blob/73959b68c25a70599adaace31154b65c82618aa9/pkgs/build-support/cc-wrapper/add-flags.sh#L38-L40

https://github.com/NixOS/nixpkgs/blob/73959b68c25a70599adaace31154b65c82618aa9/pkgs/build-support/cc-wrapper/cc-wrapper.sh#L137

Reply:
Ok good catch. Can you name it NIX_CFLAGS_COMPILE_BEFORE to mirror NIX_LDFLAGS_BEFORE?

Reply:
The reason I want this is because we also want to do a very similar thing for aarch32

Reply:
Before-vs.-after is not the only difference

```NIX_GCC_ARCH_FLAGS``` gives up to the derivation's attribute while ```NIX_CFLAGS_COMPILE``` concatenates.

Renaming ```NIX_GCC_ARCH_FLAGS```->```NIX_CFLAGS_COMPILE_BEFORE``` would make the following code very strange:

https://github.com/NixOS/nixpkgs/blob/a8fe7d80e12f41caed9866abca555f7da045423a/pkgs/development/libraries/qt-5/modules/qtwebengine.nix#L104-L109

Reply:
Oh ok. The concern I have is we wind up with 30 flags like this for each -m* option. We will also want to set flags like ```-mcpu``` ```-mfpu``` ```-mfloat-abi``` and ```-mmode```.

Reply:
One more example:

There is also a group of software intended to be copied to other machines and run there (for example ```ipxe```).
Those should be compiled with the most generic ```-march=``` irregardless of ```platform.gcc.arch``` value